testsuite/integration-tests/src/test/java/org/jboss/resteasy/microprofile/test/client/integration/RestClientNonProxyHostsPatternTest.java

```diff
@@ -0,0 +1,63 @@
+package org.jboss.resteasy.microprofile.test.client.integration;
+
+import static org.junit.Assert.assertNotNull;
+
+import java.net.URL;
+
+import org.eclipse.microprofile.rest.client.inject.RegisterRestClient;
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.container.test.api.RunAsClient;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.resteasy.microprofile.client.RestClientBuilderImpl;
+import org.jboss.resteasy.utils.PortProviderUtil;
+import org.jboss.resteasy.utils.TestUtil;
+import org.jboss.shrinkwrap.api.Archive;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * MicroProfile rest client
+ * Integration tests
+ * Testing that using nonProxyHosts property with * near index 0 doesn't throw
+ * regex.PatternSyntaxException while trying to build a rest client
+ * https://issues.redhat.com/browse/JBEAP-25126
+ */
+@RunWith(Arquillian.class)
+@RunAsClient
+public class RestClientNonProxyHostsPatternTest {
+
+    //asterisk must be at index 0
+    static final String NON_PROXY_HOSTS = "*.test.com";
+
+    @Deployment
+    public static Archive<?> deploy() {
+        WebArchive war = TestUtil.prepareArchive(RestClientNonProxyHostsPatternTest.class.getSimpleName());
+        war.addClass(HelloClient.class);
+        return TestUtil.finishContainerPrepare(war, null);
```

Comment:
We use a different utility for creating the deployment here. This should be something like:
```java
@Deployment
public static Archive<?> deploy() {
    return TestEnvironment.createWar(RestClientNonProxyHostsPatternTest.class)
            .addClasses(HelloClient.class);
}
```